source/4_tools/regina/src/regina.cpp

```diff
@@ -1,54 +1,865 @@
+        return true;
+    }
+
+    //-------------------------------------------------------------------------
+    void update_view()
+    {
+        // Convert Spherical to Cartesian coordinates.
+        g_regina_ctx.eye_pos.x = 15.0f * sinf(0.2f * glm::pi<f32>()) * cosf(1.5f * glm::pi<f32>());
+        g_regina_ctx.eye_pos.y = 15.0f * cosf(0.2f * glm::pi<f32>());
+        g_regina_ctx.eye_pos.z = 35.0f * sinf(0.2f * glm::pi<f32>()) * sinf(1.5f * glm::pi<f32>());
+
+        // Build the view matrix.
+        glm::vec3 pos = glm::vec3(g_regina_ctx.eye_pos.x, g_regina_ctx.eye_pos.y, g_regina_ctx.eye_pos.z);
+        glm::vec3 target = glm::vec3(0.0f, 0.0f, 0.0f);
+        glm::vec3 up = glm::vec3(0.0f, 1.0f, 0.0f);
+
+        g_regina_ctx.view = glm::lookAt(pos, target, up);
+        g_regina_ctx.view = glm::transpose(g_regina_ctx.view); // DirectX backend ( so we have to transpose, expects row major matrices )
+    }
+    //-------------------------------------------------------------------------
+    void update_object_constant_buffers()
+    {
+        for (auto& ri : (*g_regina_ctx.scene))
+        {
+            // Only update the cbuffer data if the constants have changed.  
+            // This needs to be tracked per frame resource.
+            if (ri.num_frames_dirty > 0)
+            {
+                // Assign the new world matrixz
+                ObjectConstants obj_constants;
+                obj_constants.world = glm::transpose(ri.world); // DirectX backend ( so we have to transpose, expects row major matrices )
+
+                renderer::commands::UpdateCommitedResourceCommandDesc update_constant_buffer_command_desc;
+                update_constant_buffer_command_desc.element_index = ri.constant_buffer_index;
+                update_constant_buffer_command_desc.buffer_data = memory::make_blob((rsl::byte*)&obj_constants, rsl::memory_size(sizeof(ObjectConstants)));
+
+                renderer::ResourceSlot curr_object_cr = get_active_object_commited_resource_for_frame(renderer::active_frame()->slot_id());
+                renderer::update_commited_resource(rsl::move(update_constant_buffer_command_desc), curr_object_cr);
+
+                // Updating constant buffer of the cube so the frame should remain dirty
+                ri.num_frames_dirty = renderer::num_frames_in_flight();
+            }
+        }
+    }
+    //-------------------------------------------------------------------------
+    void update_pass_constant_buffers()
+    {
+        renderer::ResourceSlot curr_pass_cr = get_active_pass_commited_resource_for_frame(renderer::active_frame()->slot_id());
+
+        const glm::mat4& view = g_regina_ctx.view;
+        const glm::mat4& proj = g_regina_ctx.proj;
+
+        glm::mat4 view_proj = view * proj;
+
+        glm::mat4 inv_view = glm::inverse(view);
+        glm::mat4 inv_proj = glm::inverse(proj);
+        glm::mat4 inv_view_proj = glm::inverse(view_proj);
+
+        g_regina_ctx.pass_constants.view = view;
```

Comment:
lots of C like coding here, there might be a more C++ like way of coding this.
you're also accessing members of members which usually, isn't a good thing

Reply:
I'm just assigning values to a data container. I don't think that's a bad thing. It's in line with the guidelines " structs " should contain only data once they contain functions they should become a class in this case there are no functions no operations are done on the members except assigning values. Feels like a lot of boiler plate has to be added if this turns into a class